modules/upgrading-highly-available-quay-operator.adoc

```diff
@@ -0,0 +1,92 @@
+:_content-type: PROCEDURE
+[id="upgrading-highly-available-quay-operator"]
+= Upgrading a highly available deployment of the {productname} Operator
+
+Use the following procedure to upgrade your highly available {productname} Operator.
+
+[IMPORTANT]
+====
+* When upgrading highly available {productname} Operator deployments to the next y-stream release (for example, {productname} 3.7 -> {productname} 3.8), you must stop operations before upgrading.
+* There is intermittent downtime down upgrading from one y-stream release to the next.
+* It is highly recommended to back up your {productname} Operator deployment before upgrading.
+====
+
+.Procedure
+
+[NOTE]
+====
+This procedure assumes that you are running the {productname} Operator on three (or more) systems. For this procedure, we will assume three systems named `System A,` `System B,` and `System C`. `System A` will serve as the primary system in which the {productname} Operator is deployed.
+====
+
+. On System B and System C, scale down your {productname} Operator deployment. This is done by disabling auto scaling and overriding the replica county for {productname}, mirror workers, and Clair (if it is managed). Use the following `quayregistry.yaml` file as a reference:
+[source,yaml]
+----
+apiVersion: quay.redhat.com/v1
+kind: QuayRegistry
+metadata:
+  name: registry
+  namespace: ns
+spec:
+  components:
+    …
+    - kind: horizontalpodautoscaler
+      managed: false <1>
+    - kind: quay
+      managed: true
+      overrides: <2>
+        replicas: 0
+    - kind: clair
+      managed: true
+      overrides:
+        replicas: 0
+    - kind: mirror
+      managed: true
+      overrides:
+        replicas: 0
+    …
+----
+<1> Disable auto scaling of Quay, Clair and Mirroring workers
+<2> Set the replica count to 0 for components accessing the database and objectstorage
++
+[NOTE]
+====
+You must keep the {productname} Operator running on System A. Do not update the `quayregistry.yaml` file on System A.
+====
+
+. Wait for the `registry-quay-app`, `registry-quay-mirror`, and `registry-clair-app` pods to disappear. Enter the following command to check their status:
++
+[source,terminal]
+----
+oc get pods -n <quay-namespace>
+----
++
+.Example output
++
+[source,terminal]
+----
+quay-operator.v3.7.1-6f9d859bd-p5ftc               1/1     Running     0             12m
+quayregistry-clair-postgres-7487f5bd86-xnxpr       1/1     Running     1 (12m ago)   12m
+quayregistry-quay-app-upgrade-xq2v6                0/1     Completed   0             12m
+quayregistry-quay-config-editor-6dfdcfc44f-hlvwm   1/1     Running     0             73s
+quayregistry-quay-database-859d5445ff-cqthr        1/1     Running     0             12m
```

Comment:
The doc seems to imply that Quay is deployed on 3 separate clusters but is working as one, which means Quay is running in a georeplicated environment. For georeplicated env. we need an external db, so most likely database pod will not be visible here.